cfg/conf.d/execution-time-estimator.yaml

```diff
@@ -2,17 +2,75 @@
 %TAG ! tag:fawkesrobotics.org,cfg/
 ---
 plugins/execution-time-estimator:
+  # Make sure to load the corresponding plugins when using special estimators
+  # (the config estimator is loaded by default)
+  #
+  # common options for each estimator:
+  #  priority: order in which different executors get queried
+  #  speed: control the execution speed through applying a factor on each
+  #         individual estimator
+  #  blacklist: do not use executor for blacklisted skills
+  #  whitelist: only use executor for whitelisted skills, if unspecified all
+  #             skills are whitelisted
+
+  # Reads skill execution times from this config
+  # Does not work with empty whitelist
   static:
     priority: -1
-    default: 1
-    skills:
-      say: 2.5
+    speed: 1
+
+    # global whitelist options
+    # The static estimator may also provide default execution times even for
+    # entries not mentioned in the whitelist when specifying the exec-time
+    # globally
+    exec-time: 3
+
+    # whitelist and blacklist
+    whitelist:
+      # matches all "say" skills if the other entries do not match
+      # (determined by lexicographical order)
+      99-default-say:
+        name: say
+        exec-time: 12
+      # matches, if "hello world" has to be said
+      01-say-hello-world:
+        name: say
+        args: ["text=hello world"]
+        exec-time: 17
```

Comment:
Did you test this? I would have expected that `hello world` needs to be quoted, it's a string after all.

Reply:
> Is there a reason why you don't put the script into `etc/scripts`? This would also take care of installing it automatically.

I oriented myself on the plugin assisting tool of the `mongodb` (there it is indeed a c++ tool so it makes sense to only compile it if the plugin itself is built) plugin, so i initially placed it in the the plugin folder. Remembering our handling of scripts in fawkes-robotino i thought about placing it in `etc/scripts`, however there was no folder for this in fawkes, yet and i assumed that there might be some reason behind that. If the `etc/scripts` folder was simply not needed as of yet, i can move the script of course :) no good reason on my side to keep it in the plugin folder (other than that this script is indeed very specific to the plugin itself).

Reply:
> Did you test this? I would have expected that `hello world` needs to be quoted, it's a string after all.
```
10:19:35.919432 FawkesMainThread: Disabling IPv6 support
10:19:37.814858 FawkesMainThread: Fawkes 1.3.0 startup complete
10:21:28.661057 SkillerSimulatorExecutionThread: BlackBoardNetworkHandler executes 'say{text="hello world"}'
10:21:28.663930 SkillerSimulatorExecutionThread: Executing 'say{text="hello world"}', will take 17.00 seconds
10:21:45.099255 SkillerSimulatorExecutionThread: BlackBoardNetworkHandler executes 'say{text="hello"}'
10:21:45.099282 SkillerSimulatorExecutionThread: Aborting execution of previous skill string 'say{text="hello world"}' for new goal
10:21:45.101129 SkillerSimulatorExecutionThread: Executing 'say{text="hello"}', will take 12.00 seconds
```

```
[tarikwork@localhost bin]$ ./skillet
Connection established
+# say{text="hello world"}
Executing: say{text="hello world"}
+# say{text="hello"}
Executing: say{text="hello"}
```

It works as you remove all `"` when parsing args: https://github.com/fawkesrobotics/fawkes/blob/edc998ce6b7eead0585b6ab2f29cd7047c4ceb0b/src/libs/execution_time_estimator/execution_time_estimator.cpp#L187-L193

Reply:
> > Is there a reason why you don't put the script into `etc/scripts`? This would also take care of installing it automatically.
> 
> I oriented myself on the plugin assisting tool of the `mongodb` (there it is indeed a c++ tool so it makes sense to only compile it if the plugin itself is built) plugin, so i initially placed it in the the plugin folder. Remembering our handling of scripts in fawkes-robotino i thought about placing it in `etc/scripts`, however there was no folder for this in fawkes, yet and i assumed that there might be some reason behind that. If the `etc/scripts` folder was simply not needed as of yet, i can move the script of course :) no good reason on my side to keep it in the plugin folder (other than that this script is indeed very specific to the plugin itself).

Yes, I think the reason is that we simply don't have any scripts yet. I think both ways are fine, I have a slight preference for putting it into `etc/scripts`.

That being said, the current script name is a bit too generic. Considering that this will be put in `bin`, the name should clearly convey what it does. What about `mongodb_skillsim_lookup.py`?

Reply:
> Yes, I think the reason is that we simply don't have any scripts yet. I think both ways are fine, I have a slight preference for putting it into `etc/scripts`.
> 
> That being said, the current script name is a bit too generic. Considering that this will be put in `bin`, the name should clearly convey what it does. What about `mongodb_skillsim_lookup.py`?

Yea the naming is poor atm, `mongodb_skillsim_lookup` sounds better. Putting it in `etc/scripts` won't install fawkes scripts when fawkes is used as submodule (the `etc/buildsys/root/linkscripts.mk` uses only `$(BASEDIR)` and hence it will not install scripts of fawkes). I guess for scripts it makes sense to install core scripts in any case, right?

Reply:
Yes, I guess that's a bug in `linkscripts.mk`.

Reply:
Fixed that in 158555237362d8de4316340e577a85bebd031189 and moved the script there in 9d7e944c29c93c7bf3de12f122544d21e64ed053.